hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -256,6 +259,7 @@ public void persistContainerSet(OutputStream out) throws IOException {
   public long takeSnapshot() throws IOException {
     TermIndex ti = getLastAppliedTermIndex();
     long startTime = Time.monotonicNow();
+    SingleFileSnapshotInfo lastSnapshot = storage.findLatestSnapshot();
```

Comment:
So this only deletes 1 previous snapshot, not all older than latest.

Reply:
Yes

Reply:
Got a question from above comments.
Just want to understand, every time when we take a snapshot, we are deleting old file means, only one snapshot file will be there. Is my understanding correct?

Reply:
Yes, but this will change.  I am planning to take up @bshashikant's suggestion.